test/cypress/integration/widgets/deployments_view_spec.ts

```diff
@@ -500,13 +502,77 @@ describe('Deployments View widget', () => {
                 .click()
                 .should('have.class', 'active')
                 .should('have.attr', 'title', 'Close map');
-            getDeploymentsViewMap();
+            getDeploymentsViewMap().within(() => {
+                cy.get('.leaflet-marker-icon').should('have.length', 1);
+                getMarkerByImageSrcSuffix('red.png').click();
+                getDeploymentsMapPopup().contains(deploymentName);
+            });
 
             getDeploymentsMapToggleButton()
                 .click()
                 .should('not.have.class', 'active')
                 .should('have.attr', 'title', 'Open map');
             getDeploymentsViewMap().should('not.exist');
         });
+
+        it('should render markers for various deployment states', () => {
+            cy.interceptSp('GET', /^\/sites.*_get_all_results=true/, {
+                statusCode: 200,
+                body: {
+                    items: [
+                        {
+                            name: 'Olsztyn',
+                            latitude: 53.7795,
+                            longitude: 20.49499
+                        },
+                        {
+                            name: 'Tel Aviv',
+                            latitude: 32.066667,
+                            longitude: 34.783333
+                        },
+                        {
+                            name: 'London',
+                            latitude: 51.509865,
+                            longitude: -0.118092
+                        },
+                        {
+                            name: 'Warsaw',
+                            latitude: 52.229676,
+                            longitude: 21.012229
+                        }
+                    ],
+                    metadata: {
+                        pagination: {
+                            offset: 0,
+                            size: 1000,
+                            total: 4
+                        }
+                    }
+                } as Stage.Types.PaginatedResponse<unknown>
+            });
+
+            useDeploymentsViewWidget({
+                routeHandler: {
+                    fixture: 'deployments/various-statuses.json'
+                },
+                configurationOverrides: {
+                    mapOpenByDefault: true
+                }
+            });
+
+            getDeploymentsViewMap().within(() => {
+                cy.get('.leaflet-marker-icon').should('have.length', 3).and('be.visible');
```

Comment:
Really cosmetic, but I think it reads better if assertions are done in the order starting from general and ending on detailed.
```suggestion
                cy.get('.leaflet-marker-icon').should('be.visible').and('have.length', 3);
```

Reply:
Good idea :+1: Thanks

When writing the code, I first wrote the `have.length` one and then added `be.visible` several minutes later, so I did not read the whole line :smile: Makes sense, I'll apply it soon

Reply:
Done